### ISSUE TITLE
VanillaEncryptionConfiguration can't be chosen

### ISSUE THREAD

**Issue by ralfstuckert** [spring-cloud/spring-cloud-commons#34]
The `VanillaEncryptionConfiguration` provided in class `org.springframework.cloud.bootstrap.encrypt.EncryptionBootstrapConfiguration` is thought to be used, if the `RsaSecretEncryptor` is not available:

```java
@ConditionalOnMissingClass(name = "org.springframework.security.rsa.crypto.RsaSecretEncryptor") 
protected static class VanillaEncryptionConfiguration { 
``` 

But the enclosing configuration class EncryptionBootstrapConfiguration  has a condition based on the availability of the RsaSecretEncryptor:

```java
@Configuration 
@ConditionalOnClass({ TextEncryptor.class, RsaSecretEncryptor.class }) 
@EnableConfigurationProperties(KeyProperties.class) 
public class EncryptionBootstrapConfiguration { 
``` 
So the `VanillaEncryptionConfiguration` will only be taken into account, if the `RsaSecretEncryptor` is both available and not available ;-)

**Comment by dsyer** [comment]
Probably not very useful then! We can remove it?

**Comment by ralfstuckert** [comment]
Remove the `VanillaEncryptionConfiguration` or the malicious condition? Since symetric encryption is recommended as the convenient encryption in the cloud config server documentation, I would find it rather useful.

I guess the point is superflous `RsaSecretEncryptor` condition on `EncryptionBootstrapConfiguration`, isn't it? Does it make any sense at this point?

**Comment by dsyer** [comment]
We use symmetric encryption in tests so that definitely works. I guess it only works if you have the rsa jar on the classpath as things stand?

**Comment by ralfstuckert** [comment]
No, I have explicitly exluded the rsa jar from the dependencies. (It was brought in as a compile dependency of the config Server)

Maybe I don't get it. I thought, the condition `@ConditionalOnClass({ TextEncryptor.class, RsaSecretEncryptor.class })` says: use this configuration only, if the RsaSecretEncryptor is on the classpath. And the condition `@ConditionalOnMissingClass(name = "org.springframework.security.rsa.crypto.RsaSecretEncryptor")` on the inner config says: use this config only, if the RsaSecretEncryptor is NOT on the classpath. Is that correct?

And if the condition on the outer config class fails (RsaSecretEncryptor not on the classpath), the inner config isn't used at all, right?

So, I thought the right solution would be to remove the RSA condition on the outer configuration class:
```java
@Configuration 
@ConditionalOnClass(TextEncryptor.class) 
@EnableConfigurationProperties(KeyProperties.class) 
public class EncryptionBootstrapConfiguration {
``` 
But if it works in your tests... maybe I'm just wrong ;-)

**Comment by dsyer** [comment]
No, you are right. You *need* RSA on the classpath as things were in 1.0.1, but you can still use symmetric encryption by providing only an `encrypt.key` value. (Fixed in 1.0.2.)

**Comment by ralfstuckert** [comment]
Ah, ok. Thank you.
